great3sims/psf.py

```diff
@@ -1393,64 +1405,64 @@ def generateFieldParameters(self, rng, field_index):
+        #   one which is choosen for expansion.  Note that the "psf_group" number is
+        #   another way to refer to the file named psfInfo[0], but using a number.
         for record in catalog:
             # select a random hdu
-            psf_number = int(math.floor(rng() * psf_count)) + 1 
-            record["psf_number"] = psf_number
-            filename = format%psf_number
-            if not os.path.exists(filename):
-                psf = pyfits.PrimaryHDU(hdus[psf_number-1].data)
+            psfInfo = self.psfInfos[int(math.floor(rng() * len(self.psfInfos)))]
+            record["psf_group"] = psfInfo[1]
+            record["psf_index"] = psfInfo[2]
+            record["psf_number"] = psfInfo[3]
+            psfInfo[4] = True
+        # write the tagged entries in the order of the psfInfos file.  This
+        # is sorted by library, minimizing the amount of file I/O required to fetch the psfs
+        dirname = os.path.dirname(self.draw_psf_src)
+        currentPsfFile = None
+        for i in range(len(self.psfInfos)):
+            psfInfo = self.psfInfos[i]
+            if psfInfo[4]:
+                if psfInfo[0] != currentPsfFile:
+                    if not currentPsfFile is None:
+                        hdus.close()
+                        del hdus
+                    currentPsfFile = psfInfo[0]
+                    filename = dirname + "/" + psfInfo[0]
+                    print "opening ", filename
+                    hdus = pyfits.open(filename)
+                print len(hdus), psfInfo[2]
+                psf = pyfits.PrimaryHDU(hdus[psfInfo[2]-1].data)
                 psf.header.append("GS_SCALE")
-                psf.header["GS_SCALE"] = 0.02
-                psf.writeto(filename)
-        if hdus: hdus.close()
+                psf.header["GS_SCALE"] = 0.2
+                psf.header["CRVAL1"] = 0.0
+                psf.header["CRVAL2"] = 0.0
+                psf.header["CUNIT1"] = "deg"
+                psf.header["CUNIT2"] = "deg"
+                psf.header["CRPIX1"] = 0
+                psf.header["CRPIX2"] = 0
+                psf.header["CD1_1"] = 5.55e-05
+                psf.header["CD1_2"] = 0.0
+                psf.header["CD2_1"] = 0.0
+                psf.header["CD2_2"] = 5.55e-05
+                format = dirname + "/psf_%d.fits"
+                filename = format%psfInfo[3]
+                psf.writeto(filename, clobber=True)
 
     def makeConfigDict(self, use_zero_index=True):
         """Routine to write the PSF-related parts of the config file used by GalSim to generate
         images."""
-        
-        index = self.draw_psf_src.find('.fits')
-        if index < 0: format = self.draw_psf_src + '_%d'
-        else:
-            format = self.draw_psf_src[:index] + '_%d' + self.draw_psf_src[index:]
-        
+        format = os.path.dirname(self.draw_psf_src) + "/psf_%d.fits"
+
         d = {
             'type':'InterpolatedImage',
+            'scale': .2,
```

Comment:
Is this the same as ``GS_SCALE``? If so, it should be defined in one place and re-used, rather than having ``0.2``s popping up in various parts of the code.